Mathlib/Algebra/Lie/CartanMatrix.lean

```diff
@@ -0,0 +1,321 @@
+the results for non-Cartan matrices should be regarded as junk.
+
+Recall that a Cartan matrix is a square matrix of integers `A` such that:
+ * For diagonal values we have: `A i i = 2`.
+ * For off-diagonal values (`i ≠ j`) we have: `A i j ∈ {-3, -2, -1, 0}`.
+ * `A i j = 0 ↔ A j i = 0`.
+ * There exists a diagonal matrix `D` over ℝ such that `D ⬝ A ⬝ D⁻¹` is symmetric positive definite.
+
+## Alternative construction
+
+This construction is sometimes performed within the free unital associative algebra
+`FreeAlgebra R X`, rather than within the free Lie algebra `FreeLieAlgebra R X`, as we do here.
+However the difference is illusory since the construction stays inside the Lie subalgebra of
+`FreeAlgebra R X` generated by `X`, and this is naturally isomorphic to `FreeLieAlgebra R X`
+(though the proof of this seems to require Poincaré–Birkhoff–Witt).
+
+## Definitions of exceptional Lie algebras
+
+This file also contains the Cartan matrices of the exceptional Lie algebras. By using these in the
+above construction, it thus provides definitions of the exceptional Lie algebras. These definitions
+make sense over any commutative ring. When the ring is ℝ, these are the split real forms of the
+exceptional semisimple Lie algebras.
+
+## References
+
+* [N. Bourbaki, *Lie Groups and Lie Algebras, Chapters 4--6*](bourbaki1968) plates V -- IX,
+  pages 275--290
+
+* [N. Bourbaki, *Lie Groups and Lie Algebras, Chapters 7--9*](bourbaki1975b) chapter VIII, §4.3
+
+* [J.P. Serre, *Complex Semisimple Lie Algebras*](serre1965) chapter VI, appendix
+
+## Main definitions
+
+  * `Matrix.ToLieAlgebra`
+  * `CartanMatrix.E₆`
+  * `CartanMatrix.E₇`
+  * `CartanMatrix.E₈`
+  * `CartanMatrix.F₄`
+  * `CartanMatrix.G₂`
+  * `LieAlgebra.E₆`
+  * `LieAlgebra.E₇`
+  * `LieAlgebra.E₈`
+  * `LieAlgebra.F₄`
+  * `LieAlgebra.G₂`
+
+## Tags
+
+lie algebra, semi-simple, cartan matrix
+-/
+
+set_option linter.uppercaseLean3 false
+
+universe u v w
+
+noncomputable section
+
+variable (R : Type u) {B : Type v} [CommRing R] [DecidableEq B] [Fintype B]
+
+variable (A : Matrix B B ℤ)
+
+namespace CartanMatrix
+
+variable (B)
+
+/-- The generators of the free Lie algebra from which we construct the Lie algebra of a Cartan
+matrix as a quotient. -/
+inductive Generators
+  | H : B → Generators
+  | E : B → Generators
+  | F : B → Generators
+#align cartan_matrix.generators CartanMatrix.Generators
+
+instance [Inhabited B] : Inhabited (Generators B) :=
+  ⟨Generators.H default⟩
+
+variable {B}
+
+namespace Relations
+
+open Function
+
+-- mathport name: exprH
+local notation "H" => FreeLieAlgebra.of R ∘ Generators.H
+
+-- mathport name: exprE
+local notation "E" => FreeLieAlgebra.of R ∘ Generators.E
+
+-- mathport name: exprF
+local notation "F" => FreeLieAlgebra.of R ∘ Generators.F
+
+-- mathport name: exprad
+local notation "ad" => LieAlgebra.ad R (FreeLieAlgebra R (Generators B))
+
+/-- The terms correpsonding to the `⁅H, H⁆`-relations. -/
```

Comment:
While we're tweaking this file, could you please to a global replace:

`correpsonding` ↦ `corresponding`